packages/opentelemetry-sdk-trace-base/src/Span.ts

```diff
@@ -69,14 +71,15 @@ export class Span implements api.Span, ReadableSpan {
     kind: api.SpanKind,
     parentSpanId?: string,
     links: api.Link[] = [],
-    startTime: api.TimeInput = hrTime()
+    startTime?: api.TimeInput
   ) {
     this.name = spanName;
     this._spanContext = spanContext;
     this.parentSpanId = parentSpanId;
     this.kind = kind;
     this.links = links;
-    this.startTime = timeInputToHrTime(startTime);
+    this._hrStartTime = hrTime();
+    this.startTime = timeInputToHrTime(startTime ?? Date.now());
```

Comment:
In principle it may happen that the process gets suspended for a few 100ms between line 81 and 82 resulting in two different start times - one used as start time one used to calc duration.

Not sure if it is worth to investigate a solution to avoid this. I could imagine that we could regularly update our own `performance.timeOrigin` instead of relying on that on from node.js.

I'm also not sure if `Date.now()` is worse performance wise as it might need to access the real time clock of the system and not just an high precision counter.

Reply:
If we regularly update timeOrigin we lose the primary benefits of the hrtime anyway if it isn't monotonic

Reply:
hrtime should be monotonic. To my understanding it only drifts away.
But yes, if the correction happens between span start/end the result my be strange.

Reply:
yeah that's what i meant. if we apply a correction regularly we might apply the correction during a span and end up with a strange result like end before start which was the primary reason to use the monotonic clock in the first place.